src/advanced-settings/data/api.test.js

```diff
@@ -0,0 +1,102 @@
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import {
+  getConfig, camelCaseObject, modifyObjectKeys, snakeCaseObject,
+} from '@edx/frontend-platform';
+import {
+  getCourseAdvancedSettings,
+  updateCourseAdvancedSettings,
+  getProctoringExamErrors,
+} from './api';
+import { convertObjectToSnakeCase } from '../../utils';
+
+jest.mock('@edx/frontend-platform', () => ({
+  getConfig: jest.fn(),
+  camelCaseObject: jest.fn(),
+  modifyObjectKeys: jest.fn(),
+  snakeCaseObject: jest.fn(),
+}));
+
+jest.mock('@edx/frontend-platform/auth', () => ({
+  getAuthenticatedHttpClient: jest.fn(),
+}));
+
+jest.mock('../../utils', () => ({
+  convertObjectToSnakeCase: jest.fn(),
+}));
+
+describe('courseSettings API', () => {
+  const mockHttpClient = {
+    get: jest.fn(),
+    patch: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    getAuthenticatedHttpClient.mockReturnValue(mockHttpClient);
+    getConfig.mockReturnValue({ STUDIO_BASE_URL: 'http://studio.test' });
+  });
+
+  describe('getCourseAdvancedSettings', () => {
+    it('should fetch and format course advanced settings', async () => {
+      const fakeData = { key: { value: 'some_value' } };
+      const camelCased = { key: { value: 'some_value' } };
```

Comment:
Can we actually return camel case, here?

```suggestion
      const camelCased = { key: { value: 'someValue' } };
```